### PR TITLE
Feat history

### PR DESCRIPTION
**Pull request by hinanshisuthar**
History Implementation
- A video is added to history if a user visit single video page
- User can delete a particular or all videos from history page

Review only following files 
image

### CODE REVIEW

File: src/services/historyService.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        }
+      );
+      history &&
+        dispatch({
+          type: "SET_HISTORY",
+          payload: history,
+        });
+    })();
+  } catch (error) {
+    console.log("Error in add to history handler", error);
```
Comment (Hamdrive) [inline src/services/historyService.js:26]: better to handle this with a throw statement, or can be improved with a toast implementation in the future

**Review by Hamdrive — APPROVED** [review]
Code written looks good
Liked the naming conventions used for components and files
can cleanup consoles but not mandatory

File: src/pages/Home/components/VIdeoCard.jsx [hunk 2]
```diff
@@ -1,24 +1,102 @@
...
+  const navigate = useNavigate();
   const modifiedTitle = title[0].toUpperCase() + title.slice(1).toLowerCase();
 
+  const watchlaterHandler = () => {
+    if (token) {
+      if (inWatchlater) {
+        removeFromWatchLater(dispatch, _id, token);
+      } else {
+        addToWatchLater(dispatch, video, token);
+        console.log("added to watchlist");
```
Comment (prachi-sahani) [inline src/pages/Home/components/VIdeoCard.jsx:36]: can be removed

**Review by prachi-sahani — COMMENTED** [review]
Code implementation looks fine
Good to merge

**Pull request merged**